### PR TITLE
SharedString docs

### PR DESCRIPTION
**Pull request by ChumpChief**
This change adds TSDocs for a subset of `SharedString`, `Sequence`, etc.  As a scoping measure, I limited the effort to the set that is directly used in the `CollaborativeTextArea` demo.

### CODE REVIEW

File: packages/dds/sequence/src/sharedString.ts [hunk 1]
```diff
@@ -14,18 +14,33 @@ import { SharedStringFactory } from "./sequenceFactory";
  * Fluid object interface describing access methods on a SharedString
  */
 export interface ISharedString extends SharedSegmentSequence<SharedStringSegment> {
+    /**
+     * Inserts the text at the postition.
+     * @param pos - The postition to insert the text at
```
Comment (tylerbutler) [inline packages/dds/sequence/src/sharedString.ts:19]:
```suggestion
     * Inserts the text at the position.
     * @param pos - The position to insert the text at
```

File: packages/dds/sequence/src/sharedString.ts [hunk 2]
```diff
@@ -14,18 +14,33 @@ import { SharedStringFactory } from "./sequenceFactory";
...
+     * @param text - The text to insert
+     * @param props - The properties of text
+     */
     insertText(pos: number, text: string, props?: MergeTree.PropertySet);
 
+    /**
+     * Inserts a marker at the postition.
+     * @param pos - The postition to insert the marker at
```
Comment (tylerbutler) [inline packages/dds/sequence/src/sharedString.ts:27]:
```suggestion
     * Inserts a marker at the position.
     * @param pos - The position to insert the marker at
```

File: packages/dds/sequence/src/sharedString.ts [hunk 3]
```diff
@@ -66,8 +79,7 @@ export class SharedString extends SharedSegmentSequence<SharedStringSegment> imp
     }
 
     /**
-     * Inserts a marker at a relative postition
-     *
+     * Inserts a marker at a relative postition.
      * @param relativePos1 - The relative postition to insert the marker at
```
Comment (tylerbutler) [inline packages/dds/sequence/src/sharedString.ts:83]:
```suggestion
     * Inserts a marker at a relative position.
     * @param relativePos1 - The relative position to insert the marker at
```

**Review by tylerbutler — APPROVED** [review]
I got tired of suggesting fixes for the "postition" typo. 😆 Just global search/replace before you merge.

**Comment by ChumpChief** [comment]
> I got tired of suggesting fixes for the "postition" typo. 😆 Just global search/replace before you merge.

Done for all non-public API surface.  There's still `findReconnectionPostition` in merge-tree `Client`, but I didn't want to include a breaking change in this PR.

**Pull request merged**